gobblin-api/src/main/java/gobblin/util/FileNameFormat.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util;
+
+/**
+ * Interface for constructing a file name using given components as String.
+ */
+public interface FileNameFormat {
```

Comment:
In that case, I don't see the purpose of this interface. I don't see what is the abstraction represented by this interface. If the implementation does not now what are the components, the only thing it can change is the separator. Why is that important?